pilot/pkg/config/kube/crdclient/cache_handler.go

```diff
@@ -44,10 +44,6 @@ type cacheHandler struct {
 }
 
 func (h *cacheHandler) onEvent(old interface{}, curr interface{}, event model.Event) error {
-	if err := h.client.checkReadyForEvents(curr); err != nil {
```

Comment:
I removed this since if we do this then we end up re-queuing *after* the sync event.

I think we probably want to wait until ready (would add it to the SyncQueue, so no complexity in each controller)